### PR TITLE
created a docker file to easily run it

### PR DESCRIPTION
**Pull request by lkwg82**
https://hub.docker.com/r/lkwg82/h2o-http2-server/

Next steps are to create a h2o organistion and link the docker hub with this repo to automatically build new releases.

I offer support.

### CODE REVIEW

File: Docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+               zlib-dev \
+               ca-certificates curl wget \
+    && rm -rf /var/cache/apk/*
+
+
+COPY h2o.conf /etc/
+
+WORKDIR /var/www/html
+
+EXPOSE 8080 8443
```
Comment (kazuho) [inline Docker/Dockerfile]: Isn't it more natural to expose port 80 / 443?

If that is the case, you should also remove `USER` from Dockerfile and use `user` directive in `h2o.conf`.

**Comment by kazuho** [comment]
Thank you for the PR.

How about setting up a separate repository under your github account that is dedicated to the purpose (i.e. docker image for alpine linux), and linking to it from https://h2o.examp1e.net/install.html?

**Comment by lkwg82** [comment]
I just investigated some possibilities to automagically trigger an docker build from any commit in this repo. And I like your idea to have a separate repository.

Would you be so kind to create one 'h2o.docker' maybe.

Thx. I'll will do the rest. Do you have an docker account? I'll create an organization for h2o and like to add you.

Cheers Lars

**Comment by kazuho** [comment]
@lkwg82 
> I just investigated some possibilities to automagically trigger an docker build from any commit in this repo. And I like your idea to have a separate repository.

:+1:

> Would you be so kind to create one 'h2o.docker' maybe.
> Thx. I'll will do the rest. Do you have an docker account? I'll create an organization for h2o and like to add you.

How about creating them under your account and have them linked from https://h2o.examp1e.net/install.html?  I am not familiar with Docker and hesitate to owe some responsibility for maintaining it.

**Comment by lkwg82** [comment]
ok, it is done.

https://hub.docker.com/r/lkwg82/h2o-http2-server/

I created a vm which has a cronjob and checks every night on new releases.

**Comment by lkwg82** [comment]
plz close when you added the link.

File: Docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+RUN apk update \
+    && apk upgrade \
+    # need for ocsp stapling \
+    && apk add -U perl openssl \
+    && apk add -U build-base \
+               cmake \
+               curl \
+               linux-headers \
+               zlib-dev \
+               ca-certificates curl wget \
```
Comment (unasuke) [inline Docker/Dockerfile]: It seem curl is duplicate.

```diff
RUN apk update \
    && apk upgrade \
    # need for ocsp stapling \
    && apk add -U perl openssl \
    && apk add -U build-base \
               cmake \
-               curl \
               linux-headers \
               zlib-dev \
               ca-certificates curl wget \
    && wget -O - $(curl -s https://api.github.com/repos/h2o/h2o/releases/latest | grep tarball | cut -d\" -f4) | tar xz \
        && cd h2o-* \
        && cmake -DWITH_BUNDLED_SSL=on . \
        && make install \
        && cd .. \
        && rm -rf h2o-* \
    && apk del build-base \
               cmake \
-               curl \
               linux-headers \
               zlib-dev \
               ca-certificates curl wget \
    && rm -rf /var/cache/apk/*
```

Comment (lkwg82) [inline Docker/Dockerfile]: Yes it was. See the new Dockerfile in the separate repository: https://github.com/lkwg82/h2o.docker/blob/master/Dockerfile

Cheers Lars

Comment (unasuke) [inline Docker/Dockerfile]: Oh, sorry.
Thank you!

**Comment by kazuho** [comment]
@lkwg82 Thank you for your contribution.  It is now listed in the top section of https://h2o.examp1e.net/install.html.

**Pull request closed without merging**